src/StatsTools/reweighting.jl

```diff
@@ -394,18 +405,238 @@ function mixed_estimator_df_progress(shift::Vector, hproj::Vector, vproj::Vector
+described in [Umrigar *et al.* (1993)](http://dx.doi.org/10.1063/1.465195). 
+Reweighting is done over `h` time steps and `length(shift) - skip` time steps are used 
+for the blocking analysis done with [`ratio_of_means()`](@ref). 
+`dτ` is the time step and `weights` a function that
+calulates the weights. See [`w_exp()`](@ref) and [`w_lin()`](@ref).
+Additional keyword arguments are passed on to [`ratio_of_means()`](@ref).
+
+Error propagation is done with [`MonteCarloMeasurements`](@ref).
+Results are returned as [`RatioBlockingResult`](@ref).
+
+See also [`mixed_estimator`](@ref), [`growth_estimator()`](@ref).
+"""
+function rayleigh_replica_estimator(
+    op_ol::Vector{Vector}, 
+    vec_ol::Vector{Vector}, 
+    shift::Vector{Vector}, 
+    h, 
+    dτ;
+    skip=0,
+    E_r::Vector=[mean(view(s, skip+1:length(s))) for s in shift],
+    weights=w_exp,
+    kwargs...
+)   
+    num_reps = length(shift)
+    pair_idx = 0
+    num = zeros(length(op_ol[1]) - skip)
+    denom = zeros(length(op_ol[1]) - skip)
+    for a in 1:num_reps, b in a+1:num_reps
+        pair_idx += 1
+        wts = if h == 0
+            ones(length(num))
+        else
+            weights(shift[a], h, dτ; E_r=E_r[a], skip) .* weights(shift[b], h, dτ; E_r=E_r[b], skip)
+        end        
+        @views num += wts .* op_ol[pair_idx][skip+1:end]
+        @views denom += wts .* vec_ol[pair_idx][skip+1:end]
+    end
+    return ratio_of_means(num, denom; kwargs...)
+end
+"""
+    rayleigh_replica_estimator(
+        df::DataFrame;
+        op_ol="Op1", 
+        vec_ol="dot", 
+        skip=0, 
+        Anorm=1,
+        kwargs...
+    ) -> r::RatioBlockingResult
+Compute the estimator of a Rayleigh quotient of operator ``\\hat{A}`` 
+(without reweighting i.e. `h = 0`)
+```math
+A_\\mathrm{est} = \\frac{\\sum_{a<b} \\sum_n \\mathbf{c}_a^{(n)} \\cdot \\hat{A} \\cdot \\mathbf{c}_b^{(n)}}
+    {\\sum_{a<b} \\sum_n \\mathbf{c}_a^{(n)} \\cdot \\mathbf{c}_b^{(n)}},
+```
+directly from a `DataFrame` returned by [`lomc!`](@ref). 
+The keyword arguments `op_ol` and `vec_ol` can be used to change the names of the relevant columns.
+The operator overlap data can be scaled by a prefactor `Anorm`.
+
+See [`AllOverlaps`](@ref).
+"""
+function rayleigh_replica_estimator(
+    df::DataFrame;
+    op_ol="Op1", 
+    vec_ol="dot", 
+    skip=0,
+    Anorm=1,
+    kwargs...
+)
+    num_reps = length(filter(startswith("norm_"), names(df)))
+    
+    shift_v = Vector[]  # no weights needed for h = 0, but correct length needed
+    for _ in 1:num_reps
+        push!(shift_v, Vector[])
+    end
+    E_r = Vector[]
+
+    vec_ol_v = Vector[]
+    op_ol_v = Vector[]
+    for a in 1:num_reps, b in a+1:num_reps
+        # push!(op_ol_v, df[:, Symbol("c$(a)_"*op_ol*"_c$(b)")] .* Anorm)
+        # push!(vec_ol_v, df[:, Symbol("c$(a)_"*vec_ol*"_c$(b)")])
+        push!(op_ol_v, Vector(getproperty(df, Symbol("c$(a)_"*op_ol*"_c$(b)"))) .* Anorm)
```

Comment:
```suggestion
        push!(op_ol_v, Vector(getproperty(df, Symbol("c$(a)_", op_ol, "_c$(b)"))) .* Anorm)
```
I suggest doing it this way, or even as `Symbol("c", a, "_", op_ol, "_c", b)`. Passing multiple arguments to `Symbol` concatenates and converts them automatically. This is good because there is less of a need for allocating intermediate strings. Also note when working with strings, it's preferable to use `string(a, b, c)` vs `a * b * c` for a similar reason.

By the way, you can use `df[!, Symbol(...)]` to get a DataFrame column without copying. It doesn't make any difference though, so you can use whichever you prefer.

Reply:
It doesn't make a difference because the call to `Vector()` makes a copy anyway?

Reply:
`getproperty(df, :symbol)` and `df[!, :symbol]` do the same thing - they don't copy. `df[:, :symbol]` makes a copy which in this case would make two copies (except if the compiler would able to avid one).